suite-native/module-add-accounts/src/screens/AddCoinDiscoveryRunningScreen.tsx

```diff
@@ -0,0 +1,104 @@
+import { useEffect, useState } from 'react';
+import { useDispatch, useSelector } from 'react-redux';
+
+import { networks, NetworkSymbol } from '@suite-common/wallet-config';
+import { VStack, Text, Spinner, SpinnerLoadingState } from '@suite-native/atoms';
+import { Screen } from '@suite-native/navigation';
+import { Translation } from '@suite-native/intl';
+import { Account } from '@suite-common/wallet-types';
+import {
+    AccountsRootState,
+    DeviceRootState,
+    selectDeviceAccountsByNetworkSymbol,
+    selectDeviceDiscovery,
+} from '@suite-common/wallet-core';
+import { toggleEnabledDiscoveryNetworkSymbol } from '@suite-native/discovery';
+import { useCoinEnabling } from '@suite-native/coin-enabling';
+
+import { useAddCoinAccount } from '../hooks/useAddCoinAccount';
+
+export type AddCoinDiscoveryRunningScreenProps = {
+    networkSymbol: NetworkSymbol;
+};
+
+export const AddCoinDiscoveryRunningScreen = ({ route }) => {
+    const { networkSymbol, flowType } = route.params;
+    const dispatch = useDispatch();
+    const accounts = useSelector((state: AccountsRootState & DeviceRootState) =>
+        selectDeviceAccountsByNetworkSymbol(state, networkSymbol),
+    );
+    const normalAccounts = accounts.filter(a => a.accountType === 'normal');
```

Comment:
these two values are needed only in context of `handleFinish` function and does not need to be calculated on every occasion. Move them inside of the function please.

Reply:
f471517693